src/components/Summary/GenericSummary.js

```diff
@@ -0,0 +1,103 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import Card from 'components/Card';
+import ErrorMessage from 'components/ErrorMessage';
+import FormStepSummary from 'components/FormStepSummary';
+import {LayoutColumn} from 'components/Layout';
+import Loader from 'components/Loader';
+import LogoutButton from 'components/LogoutButton';
+import Price from 'components/Price';
+import SummaryConfirmation from 'components/SummaryConfirmation';
+import {SUBMISSION_ALLOWED} from 'components/constants';
+
+const GenericSummary = ({
+  title,
+  submissionAllowed,
+  summaryData,
+  showPaymentInformation,
+  amountToPay,
+  showPreviousPageLink,
+  privacyInformation,
+  editStepText,
+  isLoading,
+  isAuthenticated,
+  errors,
+  onPrivacyCheckboxChange,
+  onSubmit,
+  onLogout,
+  onPrevPage = null,
+}) => {
+  const Wrapper = submissionAllowed === SUBMISSION_ALLOWED.yes ? 'form' : 'div';
+
+  if (isLoading) {
+    return (
+      <LayoutColumn>
+        <Loader modifiers={['centered']} />
+      </LayoutColumn>
+    );
+  }
+
+  return (
+    <Card title={title}>
+      {errors && errors.map(error => <ErrorMessage key={error}>{error}</ErrorMessage>)}
```

Comment:
Oh, yes... :facepalm: 
```suggestion
      {errors?.length && errors.map(error => <ErrorMessage key={error}>{error}</ErrorMessage>)}
```
because `Boolean([]) === true` :thisisfine:

Reply:
But doesn't `[].map(f)` always return `[]` regardless of what `f` is? `ErrorMessage` will never get called in that case.

Reply:
Good point, no idea what happened in my head when I wrote that comment :stuck_out_tongue: